### PR TITLE
[hazelcast-cpp-client] update to v5.2.0

### PR DESCRIPTION
**Pull request by akeles85**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by akeles85** [comment]
@microsoft-github-policy-service agree [company="hazelcast"]

**Comment by akeles85** [comment]
@microsoft-github-policy-service agree

File: ports/hazelcast-cpp-client/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hazelcast-cpp-client",
-  "version-semver": "5.1.0",
+  "version-semver": "5.2.0",
```
Comment (Adela0814) [inline ports/hazelcast-cpp-client/vcpkg.json:3]:
```suggestion
  "version": "5.2.0",
```

File: ports/hazelcast-cpp-client/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
   "description": "C++ client library for Hazelcast in-memory database.",
   "homepage": "https://github.com/hazelcast/hazelcast-cpp-client",
   "documentation": "http://hazelcast.github.io/hazelcast-cpp-client/index.html",
```
Comment (Adela0814) [inline ports/hazelcast-cpp-client/vcpkg.json:6]: Please add 'License' for `hazelcast-cpp-client`, thanks.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/hazelcast-cpp-client/portfile.cmake [hunk 3]
```diff
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO hazelcast/hazelcast-cpp-client
-    REF v5.1.0
-    SHA512 ffb7a00f432098b64e1455074843ca7d592b4ccf169f6c51234840ca13dc9f69ef424b1ace04455a80a6f93d230d44d5323baa4b95017f84c71f1e31ff320088 
+    REF v5.2.0
```
Comment (Adela0814) [inline ports/hazelcast-cpp-client/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

Comment (akeles85) [reply ports/hazelcast-cpp-client/portfile.cmake:4]: Fixed

File: ports/hazelcast-cpp-client/vcpkg.json (hunk 1 above)
Comment (akeles85) [reply ports/hazelcast-cpp-client/vcpkg.json:3]: Fixed

File: ports/hazelcast-cpp-client/vcpkg.json (hunk 2 above)
Comment (akeles85) [reply ports/hazelcast-cpp-client/vcpkg.json:6]: added

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**